From 7c53501dde3c1e8a0dcc5ccf0e8224f65a44df00 Mon Sep 17 00:00:00 2001
From: Jeevan Sai Madira <jeevansa@amd.com>
Date: Mon, 20 Feb 2023 16:57:49 +0530
Subject: [PATCH] Populating video alignment info in GstVideometa of kmsbuffer.

  DESCRIPTION :
    Since gstreamer 1.18 it is required to set alignment in GstVideoMeta.
    If it is not set, it defaults to no padding and no alignment.

Signed-off-by: Jeevan Sai Madira <jeevansa@amd.com>
---
 sys/kms/gstkmsbufferpool.c | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/sys/kms/gstkmsbufferpool.c b/sys/kms/gstkmsbufferpool.c
index 7c02aa2..a1545bd 100644
--- a/sys/kms/gstkmsbufferpool.c
+++ b/sys/kms/gstkmsbufferpool.c
@@ -160,12 +160,24 @@ gst_kms_buffer_pool_alloc_buffer (GstBufferPool * pool, GstBuffer ** buffer,
   gst_buffer_append_memory (*buffer, mem);
 
   if (priv->add_videometa) {
+    GstVideoMeta *meta;
+    GstVideoAlignment align;
+    GstStructure *config = gst_buffer_pool_get_config (pool);
     GST_DEBUG_OBJECT (pool, "adding GstVideoMeta");
 
-    gst_buffer_add_video_meta_full (*buffer, GST_VIDEO_FRAME_FLAG_NONE,
+    meta = gst_buffer_add_video_meta_full (*buffer, GST_VIDEO_FRAME_FLAG_NONE,
         GST_VIDEO_INFO_FORMAT (info),
         GST_VIDEO_INFO_WIDTH (info), GST_VIDEO_INFO_HEIGHT (info),
         GST_VIDEO_INFO_N_PLANES (info), info->offset, info->stride);
+    config = gst_buffer_pool_get_config (pool);
+    gst_video_alignment_reset (&align);
+    if (gst_buffer_pool_config_get_video_alignment (config, &align)) {
+      /* Set Alignment info in GstVideoMeta */
+      gst_video_meta_set_alignment (meta, align);
+    }
+    if (config) {
+      gst_structure_free (config);
+    }
   }
 
   return GST_FLOW_OK;
-- 
1.8.3.1

